src/App.tsx

```diff
@@ -1,14 +1,63 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './App.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterQuery } from './enums';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [filterQuery, setFilterQuery] = useState(FilterQuery.ALL);
+  const [searchQuery, setSearchQuery] = useState<null | string>(null);
+  const [selectedTodo, setSelectedTodo] = useState<null | Todo>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(error => {
+        // eslint-disable-next-line no-console
+        console.error(error);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    const preparedTodos = filterQuery === FilterQuery.ALL
+      ? todos
+      : todos.filter(todo => {
+        if (filterQuery === FilterQuery.ACTIVE) {
+          return !todo.completed;
+        }
+
+        return todo.completed;
+      });
+
+    if (searchQuery) {
+      return preparedTodos.filter(todo => (
+        todo.title.toLowerCase().includes(searchQuery.toLowerCase())
+      ));
+    }
+
+    return preparedTodos;
```

Comment:
it is hard to read, we can improve this code
```suggestion
  const search = searchQuery.toLowerCase().trim();

  const preparedTodos = todos.filter(todo => {
    const title = todo.title.toLowerCase();

    const foundTodo = title.includes(search);
    
        if (filterQuery === FilterQuery.ACTIVE) {
          return !todo.completed && foundTodo;
        }
        
        if(filterQuery === FilterQuery.COMPLETED) {
         return todo.completed  && foundTodo;
        }

        return foundTodo
      });
  });
  
  return preparedTodos
```